### PR TITLE
LCH-6222: Pq command for asymmetric paragraphs.

### PR DESCRIPTION
**Pull request by amangrover90**

### CODE REVIEW

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    parent::__construct($name);
+    $this->serviceFactory = $drupalServiceFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    parent::configure();
+    $this->setDescription('Analyses paragraphs and checks whether they are asymmetric based on site\'s default language.');
```
Comment (saxumVermes) [inline src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:50]: Can we get an alias for that, for example `ach-pq-ap` or something similar?

Comment (amangrover90) [reply src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:50]: Added.

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   *   Entity type manager.
+   *
+   * @return \Drupal\Core\Entity\ContentEntityInterface[]
+   *   Paragraph entities.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function getParagraphs(EntityTypeManagerInterface $etm) {
+    $paragraph_storage = $etm->getStorage(self::PARAGRAPH_STORAGE);
```
Comment (saxumVermes) [inline src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:133]: As far as I see if the entity type doesn't exist it throws an exception, or am I missing something?

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+/**
+ * Runs check against paragraphs which could be asymmetric and lists them down.
+ */
+class ContentHubPqAsymmetricParagraphs extends ContentHubPqCommandBase {
+
+  /**
+   * Paragraph storage.
+   */
+  const PARAGRAPH_STORAGE = 'paragraph';
```
Comment (saxumVermes) [inline src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:18]: Could you please give this a visibility?

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+   *   Entity type manager.
+   *
+   * @return array
+   *   The list of bundles.
+   *
+   * @throws \Exception
+   */
+  public function getBundles(string $entityType, EntityTypeManagerInterface $etm): array {
+    $bundleIds = [];
+    $entityDefinition = $etm->getDefinition($entityType);
```
Comment (saxumVermes) [inline src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:107]: Here you might want to use the second argument and return if it's NULL.

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    $this->setAliases(['ach-pq-ap']);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function runCommand(InputInterface $input, PqCommandResult $result): int {
+    /** @var \Drupal\Core\Entity\EntityTypeManagerInterface $etm */
+    $etm = $this->serviceFactory->getDrupalService('entity_type.manager');
+    $bundles = $this->getBundles(self::PARAGRAPH_STORAGE, $etm);
```
Comment (saxumVermes) [inline src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:60]: if this is an empty array we can return here earlier as well.

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php (hunk 2 above)
Comment (saxumVermes) [reply src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:133]: However if we already handled the bundle retrieval, here we already know if the entity type exists or not.

**Review by saxumVermes — CHANGES_REQUESTED** [review]
@amangrover90 I left one comment, please check, otherwise looks good!

**Comment by amangrover90** [comment]
Strangely I've pushed the commit fixing PR comments, but it's not showing up.

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php (hunk 2 above)
Comment (amangrover90) [reply src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:144]: Handled it in bundles method.

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php (hunk 4 above)
Comment (amangrover90) [reply src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php]: Done.

File: src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php (hunk 5 above)
Comment (amangrover90) [reply src/Command/PqCommands/ContentHubPqAsymmetricParagraphs.php:63]: Done.

**Review by saxumVermes — APPROVED** [review]
Looks good, let's roll it!

**Pull request merged**